activejob/test/jobs/final_failure_retry_job.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+require_relative "../support/job_buffer"
+require "active_support/core_ext/integer/inflections"
+
+class UnhandledError < StandardError; end
+class DefaultsError < StandardError; end
+class CustomCatchError < StandardError; end
+class DiscardableError < StandardError; end
+class CustomDiscardableError < StandardError; end
+class FinalFailureError < StandardError; end
+class ChildFinalFailureError < FinalFailureError; end
```

Comment:
Let's avoid defining top level constants in this test to not clash with others. Let's move to inside the job.